src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -147,12 +147,21 @@ export class PtyService extends Disposable implements IPtyService {
 	}
 
 	private async _expandTerminalInstance(t: ITerminalInstanceLayoutInfoById): Promise<IRawTerminalInstanceLayoutInfo<IPtyHostDescriptionDto | null>> {
-		const persistentTerminalProcess = this._throwIfNoPty(t.terminal);
-		const termDto = persistentTerminalProcess && await this._terminalToDto(t.terminal, persistentTerminalProcess);
-		return {
-			terminal: termDto ?? null,
-			relativeSize: t.relativeSize
-		};
+		try {
+			const persistentTerminalProcess = this._throwIfNoPty(t.terminal);
+			const termDto = persistentTerminalProcess && await this._terminalToDto(t.terminal, persistentTerminalProcess);
+			return {
+				terminal: termDto ?? null,
+				relativeSize: t.relativeSize
+			};
+		} catch (e) {
+			this._logService.trace(`Couldn't get layout info, a terminal was probably disconnected`, e.message);
+			// this will be filtered out and not reconnected
+			return {
+				terminal: null,
+				relativeSize: t.relativeSize
```

Comment:
Do we need to worry about fixing up relative sizes when a terminal disappears or does that happen already?

Reply:
happens already - assigns any leftover width to the last one

Reply:
https://github.com/microsoft/vscode/blob/b597d6123fb71df525dc84eb3bab18cb4ddea172/src/vs/workbench/contrib/terminal/browser/terminalTab.ts#L110-L110